src/troute-network/troute/NHDNetwork_edited.py

```diff
@@ -0,0 +1,382 @@
+from  .AbstractNetwork import AbstractNetwork
+import xarray as xr
+import pathlib
+from collections import defaultdict
+import troute.nhd_io as nhd_io
+import troute.nhd_preprocess as nhd_prep
+import pandas as pd
+import time
+from datetime import datetime, timedelta
+
+__showtiming__ = True #FIXME pass flag
+__verbose__ = True #FIXME pass verbosity
+
+def read_qlats(forcing_parameters, segment_index):
+    # STEP 5: Read (or set) QLateral Inputs
+    if __showtiming__:
+        start_time = time.time()
+    if __verbose__:
+        print("creating qlateral array ...")
+    qts_subdivisions = forcing_parameters.get("qts_subdivisions", 1)
+    nts = forcing_parameters.get("nts", 1)
+    qlat_input_folder = forcing_parameters.get("qlat_input_folder", None)
+    qlat_input_file = forcing_parameters.get("qlat_input_file", None)
+    if qlat_input_folder:
+        qlat_input_folder = pathlib.Path(qlat_input_folder)
+        if "qlat_files" in forcing_parameters:
+            qlat_files = forcing_parameters.get("qlat_files")
+            qlat_files = [qlat_input_folder.joinpath(f) for f in qlat_files]
+        elif "qlat_file_pattern_filter" in forcing_parameters:
+            qlat_file_pattern_filter = forcing_parameters.get(
+                "qlat_file_pattern_filter", "*CHRT_OUT*"
+            )
+            qlat_files = sorted(qlat_input_folder.glob(qlat_file_pattern_filter))
+
+        qlat_file_index_col = forcing_parameters.get(
+            "qlat_file_index_col", "feature_id"
+        )
+        qlat_file_value_col = forcing_parameters.get("qlat_file_value_col", "q_lateral")
+
+        qlat_df = nhd_io.get_ql_from_wrf_hydro_mf(
+            qlat_files=qlat_files,
+            #ts_iterator=ts_iterator,
+            #file_run_size=file_run_size,
+            index_col=qlat_file_index_col,
+            value_col=qlat_file_value_col,
+        )
+        qlat_df = qlat_df[qlat_df.index.isin(segment_index)]
+    elif qlat_input_file:
+        qlat_df = nhd_io.get_ql_from_csv(qlat_input_file)
+    else:
+        qlat_const = forcing_parameters.get("qlat_const", 0)
+        qlat_df = pd.DataFrame(
+            qlat_const,
+            index=segment_index,
+            columns=range(nts // qts_subdivisions),
+            dtype="float32",
+        )
+    
+    max_col = 1 + nts // qts_subdivisions
+    
+    if len(qlat_df.columns) > max_col:
+        qlat_df.drop(qlat_df.columns[max_col:], axis=1, inplace=True)
+    
+    if not segment_index.empty:
+        qlat_df = qlat_df[qlat_df.index.isin(segment_index)]
+
+    if __verbose__:
+        print("qlateral array complete")
+    if __showtiming__:
+        print("... in %s seconds." % (time.time() - start_time))
+
+    return qlat_df
+
+class NHDNetwork(AbstractNetwork):
```

Comment:
It would be helpful to review this in the context of the diff to the existing NHDNetwork.py (so don't change the file name...).  If you perfer doing this for your own development to have the original to look at, I suggest copying and renaming the original and just not adding that file to the git index.

One thing I'm not real sure of here is the potential initialization of the abstract classes members.  I'll have to dig in a little deeper and make sure that all members are initialized (they may be, it is just hard to tell from a first cut look)

Reply:
Ok thanks. I moved all the changes to NHDNetwork.py and deleted the "edited" file.